### PR TITLE
feat(Compiler): add support for `<ng-container>`

### PR DESCRIPTION
**Pull request by vicb**
`<ng-container>` is a logical container that can be used to group nodes but is not rendered in the DOM tree as a node.

`<ng-container>` is rendered as an HTML comment.

Thanks @tbosch for your help !

### CODE REVIEW

File: modules/@angular/compiler/src/view_compiler/view_builder.ts [hunk 1]
```diff
@@ -275,20 +282,53 @@ class ViewBuilderVisitor implements TemplateAstVisitor {
...
+ * Walks up the nodes while the direct parent is a container.
+ *
+ * Returns the outer container or the node itself when it is not a direct child of a container.
+ *
+ * @internal
+ */
+function _getOuterContainerOrSelf(node: CompileNode): CompileNode {
+  const view = node.view;
+
+  while (!node.parent.isNull() && (<ElementAst>node.parent.sourceAst).name === NG_CONTAINER_TAG &&
```
Comment (tbosch) [inline modules/@angular/compiler/src/view_compiler/view_builder.ts]: Create a helper function `isContainer(el: CompileElement): boolean` which contains the checks for the name and for `isNull`. Also use that in `visitElement`.

File: modules/@angular/core/test/linker/ng_container_integration_spec.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    beforeEachProviders(
+        () =>
+            [{
+              provide: CompilerConfig,
+              useValue: new CompilerConfig({genDebugInfo: true, useJit: useJit})
+            },
+             {provide: ANCHOR_ELEMENT, useValue: el('<div></div>')},
+    ]);
+
+    it('should not be rendered to the DOM tree',
```
Comment (tbosch) [inline modules/@angular/core/test/linker/ng_container_integration_spec.ts]: Rename: should be renderered as a comment

Comment (tbosch) [inline modules/@angular/core/test/linker/ng_container_integration_spec.ts]: Remove the `p` in this test

Comment (tbosch) [inline modules/@angular/core/test/linker/ng_container_integration_spec.ts]: Or rather add `and render children as siblings` to the test name

File: modules/@angular/core/test/linker/ng_container_integration_spec.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                   const children = getDOM().childNodes(el);
+                   expect(children.length).toBe(2);
+                   expect(getDOM().isCommentNode(children[0])).toBe(true);
+                   expect(getDOM().tagName(children[1])).toEqual('P');
+
+                   async.done();
+                 });
+           }));
+
+    it('should group inner nodes',
```
Comment (tbosch) [inline modules/@angular/core/test/linker/ng_container_integration_spec.ts]: Missing tests:

- should support injecting the container from children
- should support query of children from the container

**Comment by laco0416** [comment]
I want to use ngSwitch without non-semantic div tags. 
This feature supports the use-case?

**Comment by vicb** [comment]
@laco0416 yes

**Comment by laco0416** [comment]
@vicb Thanks! That's what I really wanted.

**Pull request merged**